Client/Frontend/UserContent/UserScripts/WalletEthereumProvider.js

```diff
@@ -0,0 +1,89 @@
+// Copyright 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+(function() {
+
+function post(method, payload) {
+  return new Promise((resolve, reject) => {
+    webkit.messageHandlers.$<handler>.postMessage({
+      "securitytoken": "$<security_token>",
+      "method": method,
+      "args": JSON.stringify(payload)
+    })
+    .then(resolve, (errorJSON) => {
+      try {
+        reject(JSON.parse(errorJSON))
+      } catch(e) {
+        reject(errorJSON)
+      }
+    })
+  })
+}
+
+Object.defineProperty(window, 'ethereum', {
+  configurable: true,
+  value: {
+    chainId: undefined,
+    networkVersion: undefined,
+    selectedAddress: undefined,
+    request: function (args) /* -> Promise<unknown> */  {
+      return post('request', args)
+    },
+    isConnected: function() /* -> bool */ {
+      return true;
+    },
+    enable: function() /* -> void */ {
+      return post('enable', {})
+    },
+    // ethereum.sendAsync(payload: JsonRpcRequest, callback: JsonRpcCallback): void;
```

Comment:
Does this project use JS Docs at all? Thinking that might help for defining these API types.

Reply:
No JS docs usage, these comments were just to help understand the `window.ethereum` methods we were manually defining

Reply:
Cool if JS docs aren't normally in use I don't think it makes sense to start now. Happy to leave them as is because I did find them helpful for figuring out the expected typing.